### DISCUSSION TITLE
CarPlay go-ios

### DISCUSSION THREAD

**Discussion by harshitshah4** [danielpaulus/go-ios#192] [Q&A]
Is there a way to mimic CarPlay Simulator using cli ? You can find CarPlay Simulator in Additional Tools For Code, you may find it here: https://developer.apple.com/xcode/resources/

Specifically,
1. When you install CarPlay Simulator on your Mac machine and attach a device to your Mac machine, and after configuring CarPlay, it creates a mic on the device and whatever you speak in your machine gets transferred to your device as mic input. Is there a way to achieve this using cli ?
2. On CarPlay Simulator I am able access "Favorite Contacts" of the device. Is there a way to achieve this using cli too ?

**Comment by danielpaulus** [comment]
Interesting. I'll have a look!

**Comment by danielpaulus** [comment]
I just tried looking at the connections with dproxy and yes, CarPlay has it's own service on the device.
`com.apple.carkit.service` is what powers the simulator functionality. So, yeah, totally possible to implement go-ios support for it :-)

**Comment by danielpaulus** [comment]
Should even be possible to record calls with that thing 🤔

**Comment by harshitshah4** [comment]
This is great ! Any quick way to try this on own ?

**Reply by danielpaulus** [reply]
not yet, needs a couple of hours of investigation
